contracts/interchain-token-service/src/contract/execute/interceptors.rs

```diff
@@ -0,0 +1,648 @@
+use axelar_wasm_std::{nonempty, FnExt};
+use cosmwasm_std::{Storage, Uint256};
+use error_stack::{bail, ensure, report, Result, ResultExt};
+use router_api::ChainNameRaw;
+
+use super::Error;
+use crate::state::{self, TokenDeploymentType};
+use crate::{DeployInterchainToken, InterchainTransfer, TokenConfig, TokenId, TokenInstance};
+
+pub fn subtract_supply_amount(
+    storage: &mut dyn Storage,
+    chain: &ChainNameRaw,
+    transfer: &InterchainTransfer,
+) -> Result<(), Error> {
```

Comment:
I think this would read better if you passed in the `TokenInstance`, subtracted the amount, and returned the new instance. And then the caller can save. I don't think the name and signature of this function gives a good idea of what it's doing